### PR TITLE
Use text-decoration: underline for monochrome links

### PR DESCRIPTION
**Pull request by francinen**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-react/issues/1652

To generate the underline beneath a monochromatic link, we are currently applying a border to the `:after` pseudoelement of the link component.

This approach does not support multi-line link text across all browsers and devices.

Chrome and Safari:
image

Firefox:
image

### WHAT is this pull request doing?
This PR uses `text-decoration: underline` to generate the expected style for monochromatic links. While `text-decoration` does not support transitions, it is the most robust way to ensure the underline renders appropriately across all browsers, string lengths, and contrast modes.

#### Chrome:
image

#### Safari:
image

#### Firefox:
image

#### iOS:
ios-after

####  Edge:
edge-after

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page, Banner, Button, Card, Heading, Link, Stack} from '../src';

interface State {}

export default class Playground extends React.Component<{}, State> {
  render() {
    return (
      <Page title="Playground">
        <Stack vertical>
          <Heading element="h2">Button rendered as a link</Heading>
          <Link monochrome>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Link>

          <Link>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Link>

          <Heading element="h2">Plain monochrome button</Heading>
          <Button plain monochrome>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Button>

          <Heading element="h2">Non-external link</Heading>
          <Link monochrome url="https://localhost">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Link>

          <Link url="https://localhost">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Link>

          <Heading element="h2">External link</Heading>
          <Link monochrome url="https://localhost" external>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Link>

          <Link url="https://localhost" external>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Link>

          <Card>
            <Card.Section>
              <p>
                Lorem ipsum dolor sit amet,{' '}
                <Link monochrome url="https://localhost">
                  onsectetur adipiscing elit, sed do eiusmod tempor incididunt
                  ut labore et dolore magna aliqua.
                </Link>
              </p>
            </Card.Section>
          </Card>

          <Banner>
            <p>
              Lorem ipsum dolor sit amet,{' '}
              <Link monochrome url="https://localhost">
                onsectetur adipiscing elit, sed do eiusmod tempor incididunt ut
                labore et dolore magna aliqua.
              </Link>
            </p>
          </Banner>
        </Stack>
      </Page>
    );
  }
}
```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by kaelig** [comment]
This should probably be changed as well:

https://github.com/Shopify/polaris-react/blob/67cb6e5dbe96226dec185d75e4d110a4aadf2029/src/components/Button/Button.scss#L295-L311

**Comment by francinen** [comment]
@kaelig Good catch 👍 I'll update this PR shortly

File: src/styles/shared/links.scss [hunk 1]
```diff
@@ -8,27 +8,5 @@
 }
 
 @mixin underlined-link() {
```
Comment (BPScott) [inline src/styles/shared/links.scss]: This mixin is used in two places, and now it's a single obvious property. What do you think to killing off the mixin and just using `text-decoration:underline;` directly where it's used in the Link and Button components to save a level of indirection?

**Comment by francinen** [comment]
@BPScott Good catch! Done 👍

**Comment by francinen** [comment]
While tophatting the documentation, I noticed the plain monochrome button demo is throwing an error in production:

image

**Comment by BPScott** [comment]
> I noticed the plain monochrome button demo is throwing an error in production:

That's the example we use, check the code area and [its source](https://github.com/Shopify/polaris-react/blob/master/src/components/Button/README.md#plain-monochrome-button). I believe this is because we try and make our examples based on realistic usage but yeah it feels odd to have the example for a Button be wrapped up in an InlineError.

**Pull request merged**